photon-ml/src/main/scala/com/linkedin/photon/ml/Params.scala

```diff
@@ -158,6 +160,9 @@ class Params {
    */
   var deleteOutputDirsIfExist: Boolean = false
 
+  var inputFormatType: InputFormatType = InputFormatType.AVRO
+  var featureDimension: Int = -1
```

Comment:
That's a very reasonable argument. How about that I introduce another PR right after this one to address the second-pass feature dimension counting problem? I'm thinking that we could also sort of write out a small ```_meta``` file in the input path if not presented, so that we could only do the extra pass once for the same input data. This could take a few more lines which might be a distraction to review here.